com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/services/ObjectStoreFileStorage.java

```diff
@@ -146,16 +148,12 @@ private void putBlobWithRetries(Blob blob, int retries) {
     }
 
     private int removeBlobsByFilter(Predicate<? super StorageMetadata> filter) {
-        Set<String> entriesToDelete = blobStore.list(container, new ListContainerOptions().withDetails())
+        return (int) blobStore.list(container, new ListContainerOptions().withDetails())
             .stream()
             .filter(filter)
             .map(StorageMetadata::getName)
-            .collect(Collectors.toSet());
-
-        if (!entriesToDelete.isEmpty()) {
-            blobStore.removeBlobs(container, entriesToDelete);
-        }
-        return entriesToDelete.size();
+            .peek(entry -> blobStore.removeBlob(container, entry))

```

Comment:
I'm not sure whether the `removeBlob` implementation is as fast as calling `removeBlobs` with a list of entries. Furthermore, previously it was clear that this method was doing more than it should've. Now it's less so. If you want to make it more readable, just extract the stream as it was before in a separate method.

Reply:
I looked into the implementations of BlobStore to check this. I'm not sure what concrete implementation we use for the injected BlobStore but as far as I see they all implement the removeBlobs method like this 
for (string name : blobNames) {
removeBlob(name)
}